diff --git a/dist/solid.js b/dist/solid.js
index 330452ff2dea9deb039a3c13e111292633cced58..0b4497b63214196733b7a187e1f835e0241a7cfb 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1578,7 +1578,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1625,7 +1624,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
