diff --git a/dist/solid.js b/dist/solid.js
index 4788ff31300dd5832f69682c9b34ca682eacb739..33ae3c563efc09e0f107fdac4abfc8f29c389055 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1573,7 +1573,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1620,7 +1619,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
