diff --git a/dist/solid.js b/dist/solid.js
index f8d7c6698944fb9deb3c1e0ffb77071da6fe562c..0751e68eb39091cc51236d0d6abb637dd7b08436 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1586,7 +1586,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return conditionValue();
                     }
               )
@@ -1636,7 +1635,6 @@ function Switch(props) {
               mp.keyed
                 ? conditionValue()
                 : () => {
-                    if (untrack(switchFunc)()?.[0] !== index) throw narrowedError("Match");
                     return conditionValue();
                   }
             )
