diff --git a/dist/node/chunks/dep-yUJfKD1i.js b/dist/node/chunks/dep-yUJfKD1i.js
index 57759ca8c9eb2362b67c078ea0504aafd888b44c..c5624eff5bb2469642ccaec9a4e864ca7cd737b2 100644
--- a/dist/node/chunks/dep-yUJfKD1i.js
+++ b/dist/node/chunks/dep-yUJfKD1i.js
@@ -46670,7 +46670,7 @@ async function resolvePlugins(config, prePlugins, normalPlugins, postPlugins) {
     ...isBuild ? [] : [
       clientInjectionsPlugin(config),
       cssAnalysisPlugin(config),
-      importAnalysisPlugin(config)
+      ...(config.build.modulePreload !== false ? importAnalysisPlugin(config) : null)
     ]
   ].filter(Boolean);
 }
