diff --git a/dist/node/chunks/dep-NjL7WTE1.js b/dist/node/chunks/dep-NjL7WTE1.js
index e8ab22b8512cc988241bff18d899ffdd897a7d3a..983e14481c1e90c38ec3d220b679c8f7497fafa5 100644
--- a/dist/node/chunks/dep-NjL7WTE1.js
+++ b/dist/node/chunks/dep-NjL7WTE1.js
@@ -65095,7 +65095,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       ...options.minify ? [terserPlugin(config)] : [],
       ...!config.isWorker ? [
