diff --git a/dist/node/chunks/dep-BW_D9zgJ.js b/dist/node/chunks/dep-BW_D9zgJ.js
index 87f48d0c9bf761b76cc7e96999384bb68b327674..f9da49f1716da54800e6056ac81e23933d9e5883 100644
--- a/dist/node/chunks/dep-BW_D9zgJ.js
+++ b/dist/node/chunks/dep-BW_D9zgJ.js
@@ -58824,7 +58824,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       terserPlugin(config),
       ...!config.isWorker ? [manifestPlugin(), ssrManifestPlugin(), buildReporterPlugin(config)] : [],
