diff --git a/dist/node/chunks/dep-D8YhmIY-.js b/dist/node/chunks/dep-D8YhmIY-.js
index 6e13e0db503da5e70a04d9209a938ff1862c2d48..d54850bca934416bc81a8b281907239aaa758865 100644
--- a/dist/node/chunks/dep-D8YhmIY-.js
+++ b/dist/node/chunks/dep-D8YhmIY-.js
@@ -65516,7 +65516,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       ...options.minify ? [terserPlugin(config)] : [],
       ...!config.isWorker ? [
