diff --git a/dist/node/chunks/dep-C1-ZB6nQ.js b/dist/node/chunks/dep-C1-ZB6nQ.js
index f78f95c645c9f139069570cbbf95ba9a609c9889..4cca6c3d436be252710ae43750aa5d8293254b09 100644
--- a/dist/node/chunks/dep-C1-ZB6nQ.js
+++ b/dist/node/chunks/dep-C1-ZB6nQ.js
@@ -65509,7 +65509,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       ...options.minify ? [terserPlugin(config)] : [],
       ...!config.isWorker ? [
