diff --git a/dist/client/build/register.js b/dist/client/build/register.js
index 95340c19195a56fb0ff3f9a24b00e4ed8ce08858..dc9fb67b8e1dc3ace58cf0322afbf4f2a73dacf7 100644
--- a/dist/client/build/register.js
+++ b/dist/client/build/register.js
@@ -6,6 +6,7 @@ var autoDestroy = selfDestroying === "true";
 function registerSW(options = {}) {
   const {
     immediate = false,
+    onBeginUpdate,
     onNeedRefresh,
     onOfflineReady,
     onRegistered,
@@ -77,6 +78,12 @@ function registerSW(options = {}) {
         }
       }
       wb.register({ immediate }).then((r) => {
+        if (onBeginUpdate) {
+          r?.addEventListener('updatefound', () => {
+            onBeginUpdate();
+          });
+        }
+
         if (onRegisteredSW)
           onRegisteredSW("__SW__", r);
         else
diff --git a/types/index.d.ts b/types/index.d.ts
index c2553517a12c98f4f7d1b0ef10a2dd203842d45e..ea9006e2d44617f80ed7dd51ce6dd83d16819ad0 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,5 +1,6 @@
 export interface RegisterSWOptions {
   immediate?: boolean
+  onBeginUpdate?: () => void
   onNeedRefresh?: () => void
   onOfflineReady?: () => void
   /**
