diff --git a/dist/node/chunks/dep-DiRA7B7B.js b/dist/node/chunks/dep-DiRA7B7B.js
index d404fb0d27acdc3829cafab5f720d4dc1f3b91d1..75c90621f6fdbc7d31a46f8529073d857ef24c4d 100644
--- a/dist/node/chunks/dep-DiRA7B7B.js
+++ b/dist/node/chunks/dep-DiRA7B7B.js
@@ -50219,7 +50219,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       terserPlugin(config),
       ...!config.isWorker ? [manifestPlugin(), ssrManifestPlugin(), buildReporterPlugin(config)] : [],
