diff --git a/dist/node/chunks/dep-CzJTQ5q7.js b/dist/node/chunks/dep-CzJTQ5q7.js
index ae55745c78a34eaf9bca42a9c1b82cca0d04086c..8474fe8cc2e5452d7ad7dc799c6921365f6765ec 100644
--- a/dist/node/chunks/dep-CzJTQ5q7.js
+++ b/dist/node/chunks/dep-CzJTQ5q7.js
@@ -65514,7 +65514,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       ...options.minify ? [terserPlugin(config)] : [],
       ...!config.isWorker ? [
